### PR TITLE
feat: cosign

### PR DESCRIPTION
**Pull request by c0rydoras**

### CODE REVIEW

File: .github/workflows/trivy-scan.yaml [hunk 1]
```diff
@@ -7,20 +7,71 @@ on:
...
+
+      - name: Convert trivy results to cosign-vuln
+        uses: aquasecurity/trivy-action@0.18.0
+        with:
+          image-ref: ${{ inputs.image-ref }}
+          scan-type: "convert"
+          format: "cosign-vuln"
+          output: "${{ inputs.image-ref }}.cosign.json"
+
+      - name: Login to GitHub Container Registry
```
Comment (hairmare) [inline .github/workflows/trivy-scan.yaml:69]: isn't this too late? afaict the cosign attest call above also needs access to the registry.

Comment (c0rydoras) [reply .github/workflows/trivy-scan.yaml]: yes it is, should be fixed now

File: .github/workflows/trivy-scan.yaml [hunk 2]
```diff
@@ -7,20 +7,71 @@ on:
...
+        if: ${{ inputs.sign }}
+
+      - name: Convert results to sarif
+        uses: aquasecurity/trivy-action@0.18.0
+        with:
+          image-ref: ${{ inputs.image-ref }}.json
+          scan-type: "convert"
           format: "sarif"
-          output: "${{ inputs.image-ref }}.sarif"
+          output: "{{ inputs.image-ref }}.sarif"
```
Comment (hairmare) [inline .github/workflows/trivy-scan.yaml:60]: not sure if you intended to remove this:
```suggestion
          output: "${{ inputs.image-ref }}.sarif"
```

**Review by hairmare — APPROVED** [review]
noice!

:shipit: 

LMTGTM

**Comment by hairmare** [comment]
the expired token from the pipeline is a bit confusing
 
it looks like i missed mentioning that the pipeline needs the `id-token` permission  like here: https://github.com/radiorabe/actions/blob/a5d28299fb920c0c7cfefd136685577eef1b1415/.github/workflows/release-container.yaml#L58

**Pull request merged**